### ISSUE TITLE
Hawkbit Backup Mechanism

### ISSUE THREAD

**Issue by ArashJavan** [eclipse/hawkbit#973]
Hello,
thanks for this great project. I am really new to Hawkbit and would like to know if it is possible to backup the repository-databases artifact and meta. And if yes how can I tell the hawkbit after new installation to use these repositories, instead of creating new ones? 

For example, using GitLab one can create a backup from GitLab-repo, by using the tools that are delivered with GitLab itself. so if the repository is for whatever reason not consistent, one can roll back to the backup version seamlessly. Is there any kind of backup mechanism in HawkBit?

Thanks and best regards
Arash

**Comment by schabdo** [comment]
Glad to hear that you like the project! As you mentioned there are two dimensions to take into account when it comes to backup and restore: Metadata and the artefact store. As of today hawkBit supports 5 different MySQL databases and 5 different artefact stores. Each of these bring their own tooling and mechanism how to perform a backup and restore.
So backup and restore is tightly coupled to your setup/configuration of hawkBit and how you operate it. Please don't get me wrong but I think hawkBit is not able to provide such a generic abstraction for all of them.

Long story short: Hawkbit doesn't provide such feature out of the box. Better go for the "native" tools offered by the metadata and artefact storage of your choice (e.g. mysqldump, S3 Glacier, ...)

**Comment by ArashJavan** [comment]
@schabdo thanks for your reply. Then I will the native tools for that.
